src/Oracles/ModuleFiles.hs

```diff
@@ -136,7 +141,16 @@ moduleFilesOracle = void $ do
                 let cmp f = compare (dropExtension f)
                     found = intersectOrd cmp files mFiles
                 return (map (fullDir -/-) found, mDir)
-        let pairs = sort [ (encodeModule d f, f) | (fs, d) <- result, f <- fs ]
+
+        mainpairs <- case mainIs of
```

Comment:
Could you add a comment here, explaining why `main`s are treated differently?

Reply:
Done.